utils/utils.jsx

```diff
@@ -138,32 +139,57 @@ export function getTeamRelativeUrl(team) {
     return '/' + team.name;
 }
 
+export function getChannelURL(channel, teamId) {
+    const state = store.getState();
+    let notificationURL;
+    if (channel && (channel.type === Constants.DM_CHANNEL || channel.type === Constants.GM_CHANNEL)) {
+        notificationURL = getCurrentRelativeTeamUrl(state) + '/channels/' + channel.name;
+    } else if (channel) {
+        const team = getTeam(state, teamId);
+        notificationURL = getTeamRelativeUrl(team) + '/channels/' + channel.name;
+    } else if (teamId) {
+        const team = getTeam(state, teamId);
+        const redirectChannel = getRedirectChannelNameForTeam(state, teamId);
+        notificationURL = getTeamRelativeUrl(team) + `/channels/${redirectChannel}`;
+    } else {
+        const currentTeamId = getCurrentTeamId(state);
+        const redirectChannel = getRedirectChannelNameForTeam(state, currentTeamId);
+        notificationURL = getCurrentRelativeTeamUrl(state) + `/channels/${redirectChannel}`;
+    }
+    return notificationURL;
+}
+
 export function notifyMe(title, body, channel, teamId, silent) {
-    showNotification({title,
-        body,
-        requireInteraction: false,
-        silent,
-        onClick: () => {
-            const state = store.getState();
-            window.focus();
-            if (channel && (channel.type === Constants.DM_CHANNEL || channel.type === Constants.GM_CHANNEL)) {
-                browserHistory.push(getCurrentRelativeTeamUrl(state) + '/channels/' + channel.name);
-            } else if (channel) {
-                const team = getTeam(state, teamId);
-                browserHistory.push(getTeamRelativeUrl(team) + '/channels/' + channel.name);
-            } else if (teamId) {
-                const team = getTeam(state, teamId);
-                const redirectChannel = getRedirectChannelNameForTeam(state, teamId);
-                browserHistory.push(getTeamRelativeUrl(team) + `/channels/${redirectChannel}`);
-            } else {
-                const currentTeamId = getCurrentTeamId(state);
-                const redirectChannel = getRedirectChannelNameForTeam(state, currentTeamId);
-                browserHistory.push(getCurrentRelativeTeamUrl(state) + `/channels/${redirectChannel}`);
-            }
-        },
-    }).catch(() => {
-        // Ignore the failure to display the notification.
-    });
+    // handle notifications in desktop app >= 4.3.0
+    if (UserAgent.isDesktopApp() && window.desktop && semver.gte(window.desktop.version, '4.3.0')) {
```

Comment:
since this code is added since v4.3.0 how is it not going to be greater than that?

Reply:
Yes, seems unnecessary. Here's my thinking:

1. Since this is the Webapp, it could be accessed by old and new Desktop Apps. Old apps need to skip this first case and jump to the next so the Webapp can trigger the notification since older Desktop Apps don't support sending notifications themselves.
2. While you are correct that any Desktop Apps >= 4.3 can be detected just by the `window.desktop` object being present, this is intended to be a pattern to also be used for future features in newer Desktop Apps where knowing the version may become more important. Perhaps not necessary for this release. I'm 0/5 for keeping or removing it.

Reply:
I think it makes sense to keep this code, as Dean says it's possible that older versions of the desktop app could still be used to access the webapp, and we'd want to avoid any unintended consequences.
I also think being specific allows for betting tracking and code readability, rather than just checking for the new object which isn't as clear.